nixos/modules/services/audio/mpd.nix

```diff
@@ -181,7 +193,12 @@ in {
 
       serviceConfig = {
         User = "${cfg.user}";
-        ExecStart = "${pkgs.mpd}/bin/mpd --no-daemon ${mpdConf}";
+        ExecStart = "${pkgs.mpd}/bin/mpd --no-daemon /etc/mpd.conf";
+        ExecStartPre = pkgs.writeScript "mpd-start-pre" ''
+          #!${pkgs.runtimeShell}
+          set -euo pipefail
+          cat ${mpdConf} ${cfg.credentialsFile} > /etc/mpd.conf
```

Comment:
Maybe we should add a comment to the generated `/etc/mpd.conf` to tell the user, that this file is auto-generated by NixOS and any changes will be overwritten on service restart.

Reply:
Good idea.